### PR TITLE
chore: livedemostore appetize builds

### PR DESCRIPTION
**Pull request by NQuinn27**
This PR adds:
-  publishing of livedemostore links for our Appetize builds.
- adding the livedemostore urls to our Slack notifications
- Refactors our appetize build and upload action to a composite action for reuse.

Three kinds of builds are sent to Appetize:

**Preview**
A preview build is published to Appetize for every PR. The livedemostore link will look like this: https://livedemostore.common.primer.io/ios/preview_731 - where the number at the end corresponds to the PR number.
Note that there will only be one build active per PR. The build is overwritten on each subsequent build - I.E the the only build available for a PR is the most recent one.

**Nightly**
A nightly build is is published to appetize each night along with our UI tests. The livedemostore URL looks like this: https://livedemostore.common.primer.io/ios/nightly_10.11.2023 - where the date of the build follows the `nightly_` portion.

**Release**
A release build is published to appetize along with our post-release PR merge action. The livedemostore url looks like this: https://livedemostore.common.primer.io/ios/release_2.18.0 - where the version number corresponding with the iOS SDK version is the final portion.

Screenshot 2023-11-11 at 10 47 10

### CODE REVIEW

File: Debug App/Tests/Unit Tests/Primer/PrimerBancontactCardDataManagerTests.swift [hunk 1]
```diff
@@ -11,7 +11,7 @@ import XCTest
 
 class PrimerBancontactCardDataManagerTests: XCTestCase {
     
-    private static let expectationTimeout = 1.0
+    private static let expectationTimeout = 5.0
```
Comment (NQuinn27) [inline Debug App/Tests/Unit Tests/Primer/PrimerBancontactCardDataManagerTests.swift:14]: These were failing all through Friday so I bumped up the timeout, lets see how they do now. cc @jnewc

Comment (NQuinn27) [reply Debug App/Tests/Unit Tests/Primer/PrimerBancontactCardDataManagerTests.swift:14]: Seems it was Xcode 15 that was flaky. Reverted to Xcode 14 for these (as it was before), and have a note to discuss in next chapter meeting

File: .github/workflows/build_test_upload.yml [hunk 2]
```diff
@@ -66,108 +66,52 @@ jobs:
           MATCH_KEYCHAIN_NAME: ${{ secrets.MATCH_KEYCHAIN_NAME }}
           MATCH_KEYCHAIN_PASSWORD: ${{ secrets.MATCH_KEYCHAIN_PASSWORD }}
           SOURCE_BRANCH: ${{ github.head_ref }}
-          PR_NUMBER: ${{ github.event.pull_request.number }}
-
   build-and-upload-to-appetize:
     needs: danger
     runs-on: macos-latest
-    timeout-minutes: 20
```
Comment (jnewc) [inline .github/workflows/build_test_upload.yml:74]: Do we want a more sensible time here? like 45m?

**Review by jnewc — APPROVED** [review]

**Review by borisprimer — APPROVED** [review]
great work!

File: .github/actions/appetize-build/action.yml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      run: |
+        node Report\ Scripts/appetize-failure-report-script.js createAppetizeSummaryReport ${{ inputs.source-branch }}
+    - name: Slack Notification
+      if: ${{ failure() }}
+      uses: slackapi/slack-github-action@v1.23.0
+      with:
+        channel-id: ${{ inputs.slack-channel }}
+        payload-file-path: '/var/tmp/appetize-failure-link-summary.json'
+      env:
+        SLACK_BOT_TOKEN: ${{ inputs.slack-reporter-token }}
```
Comment (semirp) [inline .github/actions/appetize-build/action.yml:127]: [nitpick]: Consider adding new line at the end of the file.

File: .github/workflows/build_test_upload.yml [hunk 4]
```diff
@@ -66,108 +66,52 @@ jobs:
...
+          appetize-api-token: ${{ secrets.APPETIZE_API_TOKEN }}
+          source-branch: ${{ github.head_ref || github.ref_name }}
+          pr-number: ${{ github.event.pull_request.number }}
+          slack-channel: ${{ secrets.SLACK_MOBILE_SDK_CHANNEL }}
+          slack-reporter-token: ${{ secrets.SLACK_REPORTER_BOT_TOKEN }}
+          github-run-id: ${{ github.run_id }}
+          run-tests: true
       - uses: peter-evans/find-comment@v2
-        if: ${{ success() }}
+        if: ${{ success()  }}
```
Comment (semirp) [inline .github/workflows/build_test_upload.yml:104]: [nitpick]: Consider removing unnecessary space.

File: .github/workflows/build_test_upload.yml [hunk 5]
```diff
@@ -66,108 +66,52 @@ jobs:
...
-        if: ${{ failure() && github.event.pull_request.base.ref == 'master' }}
-        uses: slackapi/slack-github-action@v1.23.0
-        with:
-          channel-id: ${{ secrets.SLACK_MOBILE_SDK_CHANNEL }}
-          payload-file-path: '/var/tmp/appetize-failure-link-summary.json'
-        env:
-          SLACK_BOT_TOKEN: ${{ secrets.SLACK_REPORTER_BOT_TOKEN }}
-
-
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (semirp) [inline .github/workflows/build_test_upload.yml:117]: [nitpick]: Consider adding new line at the end of the file.

File: .github/workflows/post-release-merge.yml [hunk 6]
```diff
@@ -32,3 +33,37 @@ jobs:
...
+          fastlane-password: ${{ secrets.FASTLANE_PASSWORD }}
+          match-keychain-name: ${{ secrets.MATCH_KEYCHAIN_NAME }}
+          match-keychain-password: ${{ secrets.MATCH_KEYCHAIN_PASSWORD }}
+          appetize-api-token: ${{ secrets.APPETIZE_API_TOKEN }}
+          source-branch: ${{ github.head_ref || github.ref_name }}
+          pr-number: ""
+          slack-channel: ${{ secrets.SLACK_MOBILE_SDK_CHANNEL }}
+          slack-reporter-token: ${{ secrets.SLACK_REPORTER_BOT_TOKEN }}
+          github-run-id: ${{ github.run_id }}
+          run-tests: false
```
Comment (semirp) [inline .github/workflows/post-release-merge.yml:69]: [nitpick]: Consider adding new line at the end of the file.

**Review by semirp — APPROVED** [review]

**Review by semirp — APPROVED** [review]

**Pull request merged**